packages/framework/ai-collab/src/explicit-strategy/index.ts

```diff
@@ -0,0 +1,362 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	getSimpleSchema,
+	normalizeFieldSchema,
+	Tree,
+	type ImplicitFieldSchema,
+	type SimpleTreeSchema,
+	type TreeNode,
+	type TreeView,
+} from "@fluidframework/tree/internal";
+// eslint-disable-next-line import/no-internal-modules
+import { zodResponseFormat } from "openai/helpers/zod";
+import type {
+	ChatCompletionCreateParams,
+	// eslint-disable-next-line import/no-internal-modules
+} from "openai/resources/index.mjs";
+import { z } from "zod";
+
+import type { OpenAiClientOptions, TokenUsage } from "../aiCollabApi.js";
+
+import { applyAgentEdit } from "./agentEditReducer.js";
+import type { EditWrapper, TreeEdit } from "./agentEditTypes.js";
+import { IdGenerator } from "./idGenerator.js";
+import {
+	getEditingSystemPrompt,
+	getPlanningSystemPrompt,
+	getReviewSystemPrompt,
+	toDecoratedJson,
+	type EditLog,
+} from "./promptGeneration.js";
+import { generateGenericEditTypes } from "./typeGeneration.js";
+import { fail } from "./utils.js";
+
+const DEBUG_LOG: string[] = [];
+
+/**
+ * {@link generateTreeEdits} options.
+ *
+ * @internal
+ */
+export interface GenerateTreeEditsOptions<TSchema extends ImplicitFieldSchema> {
+	openAI: OpenAiClientOptions;
+	treeView: TreeView<TSchema>;
+	treeNode: TreeNode;
```

Comment:
It would be good to document how `treeNode` and `treeView` relate here. Is `treeView` needed?